### PR TITLE
block: remove isTruthy and isFalsy

### PR DESCRIPTION
**Pull request by gabrocheleau**
Extracts block from #2233

### CODE REVIEW

File: packages/block/src/header-from-rpc.ts [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+  options?: BlockOptions
+) {
   const {
     parentHash,
     sha3Uncles,
     miner,
     stateRoot,
     transactionsRoot,
-    receiptRoot,
+    receiptRoot, // TODO: Investigate dual receiptRoot/receiptsRoot usage.
```
Comment (jochem-brouwer) [inline packages/block/src/header-from-rpc.ts:22]: This TODO can be removed, right?

File: packages/block/test/block.spec.ts [hunk 2]
```diff
@@ -176,7 +176,7 @@ tape('[Block]: block functions', function (t) {
     const common = new Common({ chain: Chain.Goerli, hardfork: Hardfork.Chainstart })
 
     try {
-      blockFromRpc(testDataFromRpcGoerli, [], { common })
+      blockFromRpc(testDataFromRpcGoerli as any, [], { common })
```
Comment (jochem-brouwer) [inline packages/block/test/block.spec.ts:179]: Can cast this as `JsonRPCBlock`?

File: packages/block/test/from-rpc.spec.ts [hunk 3]
```diff
@@ -83,9 +83,13 @@ tape('[fromRPC]:', function (t) {
     'should create a block given json data that includes a difficulty parameter of type integer string',
     function (st) {
       const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.London })
-      const blockDifficultyAsInteger = blockFromRpc(blockDataDifficultyAsInteger, undefined, {
-        common,
-      })
+      const blockDifficultyAsInteger = blockFromRpc(
+        blockDataDifficultyAsInteger as any,
```
Comment (jochem-brouwer) [inline packages/block/test/from-rpc.spec.ts:87]: Same in this file, guess those `any`s can be removed/changed?

**Review by jochem-brouwer — COMMENTED** [review]
Two points about types.

File: packages/block/test/from-rpc.spec.ts (hunk 3 above)
Comment (gabrocheleau) [reply packages/block/test/from-rpc.spec.ts:87]: Indeed! done

File: packages/block/test/block.spec.ts (hunk 2 above)
Comment (gabrocheleau) [reply packages/block/test/block.spec.ts:179]: yes! done

File: packages/block/src/header-from-rpc.ts (hunk 1 above)
Comment (gabrocheleau) [reply packages/block/src/header-from-rpc.ts]: Yep, done by merging with master

**Review by jochem-brouwer — APPROVED** [review]
If CI passes then OK! Thanks!

File: packages/block/src/header-from-rpc.ts [hunk 4]
```diff
@@ -1,15 +1,18 @@
...
 
 /**
  * Creates a new block header object from Ethereum JSON RPC.
  *
  * @param blockParams - Ethereum JSON RPC of block (eth_getBlockByNumber)
  * @param options - An object describing the blockchain
  */
-export function blockHeaderFromRpc(blockParams: any, options?: BlockOptions) {
+export function blockHeaderFromRpc(
+  blockParams: Omit<JsonRpcBlock, 'receiptsRoot'> & { receiptRoot?: string; receiptsRoot?: string },
```
Comment (acolytec3) [inline packages/block/src/header-from-rpc.ts:13]: Do we still need this `receiptRoot` in here?  I thought we decided to do away with it in #2256?  I looked at the Ethereum JSON RPC specs and it's definitely `receiptsRoot` in the [JSON-RPC definition](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getblockbyhash)

Comment (gabrocheleau) [reply packages/block/src/header-from-rpc.ts:13]: oh no we don't, messed that up during the "extraction" of the large PR into package-specific PRs, fixed

**Pull request merged**